components/ai_chat/content/browser/ai_chat_throttle.cc

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/content/browser/ai_chat_throttle.h"
+
+#include <memory>
+
+#include "brave/components/ai_chat/core/common/buildflags/buildflags.h"
+#include "brave/components/ai_chat/core/common/features.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/common/url_constants.h"
+
+namespace ai_chat {
+
+// static
+std::unique_ptr<AiChatThrottle> AiChatThrottle::MaybeCreateThrottleFor(
+    content::NavigationHandle* navigation_handle) {
+  if (!ai_chat::features::IsAIChatEnabled()) {
+    return nullptr;
+  }
+
+  // We need this throttle to work only for chrome-untrusted://chat page
+  if (!navigation_handle->GetURL().SchemeIs(
+          content::kChromeUIUntrustedScheme) ||
+      navigation_handle->GetURL().host_piece() != kChatUIHost) {
+    return nullptr;
+  }
+
+  // Purpose of this throttle is to forbid loading of chrome-untrusted://chat
```

Comment:
Checking the page transition type does not seem like a good way to prevent this. Just get the `WebContents` from the navigation handle and call `sessions::SessionTabHelper::IdForTab`. A non-tab will have `SessionID::InvalidValue`